products.php

```diff
@@ -6,9 +6,21 @@
 require_once 'src/Models/CategoryModel.php';
 require_once 'src/Entities/Product.php';
 
+$categoryTitle = 'Oops! Invalid product ID';
+$productList = [];
 $db = PdoFactory::connect();
+$NumberOfCategories = ProductModel::getNumberOfCategories($db);
 
-$productList = ProductModel::getProductList($_GET['cat_id'], $db);
+if (isset($_GET['cat_id']) && is_numeric($_GET['cat_id'])) {
+    if ($_GET['cat_id'] <= $NumberOfCategories && $_GET['cat_id'] > 0) {
```

Comment:
this if statement seems unnecesary? if there are no products within a category the DB will return false, you dont need to check this first